client.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2021 Snowflake Computing Inc. All right reserved.
+
+package gosnowflake
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync/atomic"
+	"time"
+)
+
+// Client holds an InternalClient
+type Client struct {
+	client InternalClient
+}
+
+// NewClient returns a Client
+func NewClient() *Client {
+	cli := &HTTPClient{}
+	return &Client{cli}
+}
+
+// InternalClient is implemented by HTTPClient
+type InternalClient interface {
+	Get(string, map[string]string, time.Duration) (*http.Response, error)
+	Post(string, map[string]string, []byte, time.Duration) (*http.Response, error)
+}
+
+// HTTPClient implements InternalClient
+type HTTPClient struct {
+	cfg *Config
+	st  http.RoundTripper
+}
+
+// SetConfig sets config
+func (cli *HTTPClient) SetConfig(config *Config) {
+	cli.cfg = config
+	cli.st = SnowflakeTransport
+	if config.Transporter == nil {
+		if config.InsecureMode {
+			cli.st = snowflakeInsecureTransport
+		} else {
+			ocspResponseCacheLock.Lock()
+			atomic.StoreUint32((*uint32)(&ocspFailOpen), uint32(config.OCSPFailOpen))
+			ocspResponseCacheLock.Unlock()
```

Comment:
I see we do `lock()` and then `unlock()` but in between we already use `atomic`, so is it still necessary to do the mutex locking if the storeUint32 is already atomic?

Reply:
I had thought so too, but looking at this: https://github.com/snowflakedb/gosnowflake/blob/master/driver.go#L45, I thought it would be better safe than sorry.